logcheck/plugin/plugin.go

```diff
@@ -19,17 +19,58 @@ limitations under the License.
 package main
 
 import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
 	"golang.org/x/tools/go/analysis"
 	"sigs.k8s.io/logtools/logcheck/pkg"
 )
 
 type analyzerPlugin struct{}
 
 func (*analyzerPlugin) GetAnalyzers() []*analysis.Analyzer {
-	return []*analysis.Analyzer{
-		pkg.Analyser(),
-	}
+	analyzer, _ := pkg.Analyser()
+	return []*analysis.Analyzer{analyzer}
 }
 
 // AnalyzerPlugin is the entry point for golangci-lint.
 var AnalyzerPlugin analyzerPlugin
+
+type settings struct {
+	Check  map[string]bool `json:"check"`
+	Config string          `json:"config"`
+}
+
+// New API, see https://github.com/golangci/golangci-lint/pull/3887.
+func New(pluginSettings interface{}) ([]*analysis.Analyzer, error) {
+	// We could manually parse the settings. This would involve several
+	// type assertions. Encoding as JSON and then decoding into our
+	// settings struct is easier.
+	//
+	// The downside is that format errors are less user-friendly.
+	var buffer bytes.Buffer
+	if err := json.NewEncoder(&buffer).Encode(pluginSettings); err != nil {
```

Comment:
If we don't mind adding a third dependency of "github.com/mitchellh/mapstructure", it's simpler the parse `pluginSettings`.

Reply:
golangci-lint already depends on it, so it wouldn't make a difference for Kubernetes in `hack/tools`. But here I prefer to avoid such a heavy dependency.

I am also not sure whether mapstructure would provide better error messages. Does it?

Reply:
mapstructure provides detailed error messages but I think it is most useful for decoding purposes.